### DISCUSSION TITLE
Metalink

### DISCUSSION THREAD

**Discussion by sjehuda** [curl/curl#15818] [General]
Greetings!

This discussion thread is for the purpose to coordinate efforts to attempt to restore Metalink support.

See
- https://github.com/curl/curl/pull/7176
- https://github.com/metalink-dev/libmetalink/issues/5

**Comment by sjehuda** [comment]
> > >Metalink requires an XML parsing library, which is complex code (even the smaller alternatives) and to this day often gets security updates.

> > Please elaborate, so I would know better about this specific matter.

> What about it is not clear?

@bagder.

You have referred to https://github.com/metalink-dev/libmetalink

I *suppose*, that XML should not be a task to handle with cURL.

Did cURL utilize both an XML library and a Metalink library to handle Metalinks?

What was the procedure?

**Reply by bagder** [reply]
The question is rather what you need to do now to bring metalink support to curl:

You need to convince us that adding metalink to curl is a better idea than you writing your own separate tool for metalink transfers (possibly using libcurl). Metalink was always an awkward and strange match for curl since it took the tool a step away from "download the URL provided on the command line".

The metalink and XML libraries etc that you need to add to this are just two additional complicating matters.

The metalink support was documented. You can check out curl from before 265b14d6b37c4298bd5556fabcbc37d36f911693 and read the docs to learn how it was used.

**Reply by sjehuda** [reply]
Thank you. I will.

I will also further contemplate about the cause of cURL aganist Metalink.

> "download the URL provided on the command line".

**Comment by sjehuda** [comment]
> > I plea to reconsider Metalink

> Given the number of problems and challenges involved I think it is unlikely that we would accept a PR bringing it back, but if everything was done properly, documented, tested etc then maybe. I am not at all convinced anyone is willing and able to do what's necessary to make such a PR.

Please advise on how can Metalink support be restored and **properly** added.

As you determine to be most proper fit.
